android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/SearchExtension.kt

```diff
@@ -47,81 +48,76 @@ fun Search.filterByResourceTypeId(
 }
 
 fun Search.filterBy(dataQuery: DataQuery) {
-  when (dataQuery.filterType) {
-    Enumerations.SearchParamType.TOKEN -> filterToken(dataQuery)
-    Enumerations.SearchParamType.STRING -> filterString(dataQuery)
-    Enumerations.SearchParamType.DATE -> filterDate(dataQuery)
-    else ->
-      throw UnsupportedOperationException("Can not apply ${dataQuery.filterType} as search filter")
+  dataQuery.filterCriteria.forEach { filterCriterion: FilterCriterionConfig ->
+    when (filterCriterion) {
```

Comment:
do the conditions cover the possible types of `filterCriterion`? if not, is it better to ignore the filter (as these changes do) or throw?

Reply:
The conditions cover all the possible types.